### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by AntonNaumchik**

### CODE REVIEW

File: HW_03/Anton_Naumchuk/hw_03_t_02.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# rubocop:disable Metrics/AbcSize
```
Comment (labs-ruby) [inline HW_03/Anton_Naumchuk/hw_03_t_02.rb]: Do not disable rubocop, create separate methods if needed

Comment (AntonNaumchik) [reply HW_03/Anton_Naumchuk/hw_03_t_02.rb]: > Do not disable rubocop, create separate methods if needed

already fix

File: HW_03/Anton_Naumchuk/hw_03_t_01.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+def task_1(str)
+  array = str.split("\n")
```
Comment (labs-ruby) [inline HW_03/Anton_Naumchuk/hw_03_t_01.rb]: str.lines

File: HW_03/Anton_Naumchuk/hw_03_t_02.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+DATE_REGEXP = %r{\w{2}/\w{3}/\d{4}:\d{2}:\d{2}:\d{2}\D+\d{4}}.freeze
+IP_REGEXP = /\d+.\d+.\d+.\d+/.freeze
+
+def task_2(str)
+  str.split("\n")
+  addr = str.upcase.scan(ADDRESS_REGEXP)
+  date = str.scan(DATE_REGEXP)
+  ip = str.scan(IP_REGEXP)
+  pp = str_1(date, addr, ip), str_2(date, addr, ip), str_3(date, addr, ip)
+  addr.to_s.length < 5 ? '' : pp
```
Comment (labs-ruby) [inline HW_03/Anton_Naumchuk/hw_03_t_02.rb]: What is this magic number? -5

Comment (labs-ruby) [reply HW_03/Anton_Naumchuk/hw_03_t_02.rb]: set var .... = 5

File: HW_03/Anton_Naumchuk/hw_03_t_02.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+end
+
+def str_1(date, addr, ip)
+  "#{date[0]} FROM: #{ip[0]} TO: /#{addr[0]}"
```
Comment (labs-ruby) [inline HW_03/Anton_Naumchuk/hw_03_t_02.rb]: date.first

File: HW_03/Anton_Naumchuk/hw_03_t_02.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+end
+
+def str_2(date, addr, ip)
+  "#{date[1]} FROM: #{ip[3]} TO: /#{addr[1]}"
+end
+
+def str_3(date, addr, ip)
+  "#{date[2]} FROM: #{ip[3]} TO: /#{addr[2]}"
```
Comment (labs-ruby) [inline HW_03/Anton_Naumchuk/hw_03_t_02.rb]: date.last

File: HW_03/Anton_Naumchuk/hw_03_t_03.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+def task_3(str)
+  msg = 'Calling core with action'
+  arr = []
+  str.each_line { |x| arr << Time.parse(x.chomp) if x.include?(msg) }
+  arr.size == 2 ? (arr[1] - arr[0]).to_s : '0'
```
Comment (labs-ruby) [inline HW_03/Anton_Naumchuk/hw_03_t_03.rb]: If there are several errors, more than 2? - will return incorrect result?

**Pull request merged**